ocs_ci/ocs/platform_nodes.py

```diff
@@ -217,14 +217,25 @@ def get_vms(self, nodes):
             list: vSphere vm objects list
 
         """
-        vms_in_pool = self.vsphere.get_all_vms_in_pool(
-            self.cluster_name, self.datacenter, self.cluster
-        )
-        node_names = [node.get().get("metadata").get("name") for node in nodes]
+        vms_all = []
         vms = []
+        logger.info(f"deployment type: {self.deployment_type}")
+        if self.deployment_type == "upi":
+            vms_all = self.vsphere.get_all_vms_in_pool(
+                self.cluster_name, self.datacenter, self.cluster
+            )
+        elif self.deployment_type == "ipi":
+            vms_all = self.vsphere.get_all_vms_in_dc(self.datacenter)
+
+        else:
+            logger.error("Deployment type not detected")
```

Comment:
I think we should return None here if deployment type is not detected

Reply:
done

Reply:
does the test process None if passed?

Reply:
there are some tests that use this function indirectly, it's a really "inner" function. they should fail. 
but this less likely should happen because a stable and correct VMware deployment will have no problem returning deployment type. else is there just to match the if/elif/else

Reply:
ack